lib/spack/llnl/util/compat.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+# isort: off
+
+import multiprocessing
+import sys
+
+try:
+    from collections import OrderedDict as OrderedDict
+except ImportError:
+    try:
+        from collections.abc import (    # type: ignore[attr-defined,no-redef]
+            OrderedDict as OrderedDict,
+        )
+    except ImportError:
+        from ordereddict import (        # type: ignore[no-redef] # noqa: F401
+            OrderedDict as OrderedDict,
+        )
+
+if sys.version_info < (3,):
+    from itertools import ifilter as filter
+    from itertools import imap as map
+    from itertools import izip as zip
+    from itertools import izip_longest as zip_longest
+    from urllib import urlencode as urlencode
+    from urllib import urlopen as urlopen
+else:
+    filter = filter
+    map = map
+    zip = zip
+    from itertools import zip_longest as zip_longest  # novm # noqa: F401
+    from urllib.parse import urlencode as urlencode   # novm # noqa: F401
+    from urllib.request import urlopen as urlopen     # novm # noqa: F401
+
+if sys.version_info >= (3, 3):
+    from collections.abc import Hashable as Hashable                # novm
+    from collections.abc import Iterable as Iterable                # novm
+    from collections.abc import Mapping as Mapping                  # novm
+    from collections.abc import MutableMapping as MutableMapping    # novm
+    from collections.abc import MutableSequence as MutableSequence  # novm
+    from collections.abc import MutableSet as MutableSet            # novm
+    from collections.abc import Sequence as Sequence                # novm
+else:
+    from collections import Hashable as Hashable                # noqa: F401
+    from collections import Iterable as Iterable                # noqa: F401
+    from collections import Mapping as Mapping                  # noqa: F401
+    from collections import MutableMapping as MutableMapping    # noqa: F401
+    from collections import MutableSequence as MutableSequence  # noqa: F401
+    from collections import MutableSet as MutableSet            # noqa: F401
+    from collections import Sequence as Sequence                # noqa: F401
+
+# On macOS, Python 3.8 multiprocessing now defaults to the 'spawn' start
+# method. Spack cannot currently handle this, so force the process to start
+# using the 'fork' start method.
+#
+# TODO: This solution is not ideal, as the 'fork' start method can lead to
+# crashes of the subprocess. Figure out how to make 'spawn' work.
+#
+# See:
+# * https://github.com/spack/spack/pull/18124
+# * https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
+# * https://bugs.python.org/issue33725
+if sys.version_info >= (3, 8):
+    fork_context = multiprocessing.get_context('fork')
+else:
+    fork_context = multiprocessing
```

Comment:
I didn't think we were still doing this. Didn't we properly define pickling procedures for all objects that needed it?

Reply:
Yes, it looks like this is no longer used anywhere. I will remove it!